eox_tagging/api/v1/viewset.py

```diff
@@ -0,0 +1,54 @@
+"""
+Viewset for Tags.
+"""
+from django_filters import rest_framework as filters
+from rest_framework import viewsets
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAdminUser
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from eox_tagging.api.v1.filters import TagFilter
+from eox_tagging.api.v1.pagination import TagApiPagination
+from eox_tagging.api.v1.serializers import TagSerializer
+from eox_tagging.edxapp_wrappers import get_site
+from eox_tagging.models import Tag
+
+
+class TagViewSet(viewsets.ModelViewSet):
+    """Viewset for listing and creating Tags."""
+
+    serializer_class = TagSerializer
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (IsAdminUser,)
+    pagination_class = TagApiPagination
+    filter_backends = (filters.DjangoFilterBackend,)
+    filter_class = TagFilter
+    lookup_field = "key"
+    http_method_names = ["get", "post", "delete", "head"]
+
+    def get_queryset(self):
+        """Restricts the returned tags."""
+        owner_type = self.request.query_params.get('owner_type', None)
+        include_invalid = self.request.query_params.get('include_invalid', None)
+        user = self.request.user
+        site = get_site()
+
+        if include_invalid in ["true", "1"]:
+            queryset = Tag.objects.all()
+        else:
+            queryset = Tag.objects.valid()
+
+        if owner_type:
```

Comment:
why don't set to general parameters  and execute that once 
for example 
`type = owner_type if owner_type == "user" else 'site'`
`id = user.username if owner_type == "user" else site.id`
inside the try only this 
`return queryset.find_by_owner(owner_type=type, owner_id=id)`

I don't understand the line 50, this `queryset.find_by_owner(owner_type="user", owner_id=user.username)` will be the default value  ?

if the answer is yes `return queryset.find_by_owner(owner_type=type, owner_id=id) | queryset.find_by_owner(owner_type="user", owner_id=user.username)`

you will need only one try except

Reply:
Nice!! I'll try that